nilearn/_utils/data_gen.py

```diff
@@ -866,91 +878,437 @@ def create_fake_bids_dataset(base_dir='',
-        os.makedirs(bids_path)
-        for subject in ['sub-%02d' % label for label in range(1, 11)]:
-            for session in created_sessions:
-                subses_dir = os.path.join(bids_path, subject, session)
-                func_path = os.path.join(subses_dir, 'func')
-                os.makedirs(func_path)
-                for task, n_run in zip(tasks, n_runs):
-                    for run in [
-                            'run-%02d' % label
-                            for label in range(1, n_run + 1)
-                    ]:
-                        fields = [subject, session, 'task-' + task]
-                        if '' in fields:
-                            fields.remove('')
-                        file_id = '_'.join(fields)
-                        if n_run > 1:
-                            file_id += '_' + run
-                        preproc = (
-                            file_id + '_space-MNI_desc-preproc_bold.nii.gz')
-                        preproc_path = os.path.join(func_path, preproc)
-                        write_fake_bold_img(preproc_path, [vox, vox, vox, 100],
-                                            random_state=rand_gen)
-                        preproc = (
-                            file_id + '_space-T1w_desc-preproc_bold.nii.gz')
-                        preproc_path = os.path.join(func_path, preproc)
-                        write_fake_bold_img(preproc_path, [vox, vox, vox, 100],
-                                            random_state=rand_gen)
-                        preproc = (
-                            file_id + '_space-T1w_desc-fmriprep_bold.nii.gz')
-                        preproc_path = os.path.join(func_path, preproc)
-                        write_fake_bold_img(preproc_path, [vox, vox, vox, 100],
-                                            random_state=rand_gen)
-                        if with_confounds:
-                            confounds_path = os.path.join(
-                                func_path,
-                                file_id + '_' + confounds_tag + '.tsv',
-                            )
-                            basic_confounds(100, random_state=rand_gen).to_csv(
-                                confounds_path, sep='\t', index=None)
-    return 'bids_dataset'
+        _mock_bids_derivatives(bids_path=Path(base_dir) / bids_dataset_dir,
+                               n_sub=n_sub,
+                               n_ses=n_ses,
+                               tasks=tasks,
+                               n_runs=n_runs,
+                               no_session=no_session,
+                               with_confounds=with_confounds,
+                               confounds_tag=confounds_tag,
+                               entities=entities,
+                               n_voxels=n_voxels,
+                               rand_gen=rand_gen)
+
+    return bids_dataset_dir
+
+
+def _is_alphanumeric(string: str) -> bool:
```

Comment:
`_is_alphanumeric` -> `is_alphanumeric`

It is not a limited function and can be called all over by any code; also it is in the `_utils` package, which is non user-facing.

Of course, a non-blocker, stylistic stuff.

Reply:
ok maybe extracting a function for that is too much

will inline it instead